allensdk/brain_observatory/behavior/data_objects/stimuli/presentations.py

```diff
@@ -540,17 +542,37 @@ def _postprocess(
                     )
                 }
             )
-        # Check if the first entry in the DataFrame is an omitted stimulus.
-        # This shouldn't happen and likely reflects some sort of camstim error
-        # with appending frames to the omitted flash frame log. See
-        # explanation here:
-        # https://github.com/AllenInstitute/AllenSDK/issues/2577
-        if "omitted" in df.columns and len(df) > 0:
-            first_row = df.iloc[0]
+        df = cls._check_for_errant_omitted_stimulus(input_df=df)
+        return df
+
+    @staticmethod
+    def _check_for_errant_omitted_stimulus(
+        input_df: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Check if the first entry in the DataFrame is an omitted stimulus.
+
+        This shouldn't happen and likely reflects some sort of camstim error
+        with appending frames to the omitted flash frame log. See
+        explanation here:
+        https://github.com/AllenInstitute/AllenSDK/issues/2577
+
+        Parameters
+        ----------/
+        input_df : DataFrame
+            Input stimulus table to check for "omitted" stimulus.
+
+        Returns
+        -------
+        modified_df : DataFrame
+            Dataframe with omitted stimulus removed from first row or if not
+            found, return input_df unmodified.
+        """
+        if "omitted" in input_df.columns and len(input_df) > 0:
+            first_row = input_df.iloc[0]
             if not pd.isna(first_row["omitted"]):
                 if first_row["omitted"]:
-                    df = df.drop(first_row.name, axis=0)
-        return df
+                    input_df.drop(first_row.name, axis=0, inplace=True)
```

Comment:
I don't see a reason for changing inplace=True for this. This method is returning the df.

Reply:
We usually just use the functions to overwrite the dataframe in the code but I've changed it back to df = df[modify.]